zarr/tests/test_convenience.py

```diff
@@ -579,42 +579,38 @@ def test_logging(self):
     h5py = None
 
 
-def temp_h5f():
-    fn = tempfile.mktemp()
-    atexit.register(os.remove, fn)
-    h5f = h5py.File(fn, mode='w')
-    atexit.register(lambda v: v.close(), h5f)
-    return h5f
+if h5py is not None:
 
-
-@pytest.mark.skipif(h5py is None, reason='h5py not installed')
```

Comment:
we also ran into this in https://github.com/pydata/xarray/pull/1557. This should work but just for reference, we created our own skipif machinery using `unittest`. 

https://github.com/pydata/xarray/blob/186a8bbe6a79e8d94203af113297c96a045b8730/xarray/tests/__init__.py#L47-L61

Reply:
Thanks, very helpful.